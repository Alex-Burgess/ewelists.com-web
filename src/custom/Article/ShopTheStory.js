import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardBody from "components/Card/CardBody.js";

import styles from "assets/jss/custom/components/shopTheStoryStyle.js";
const useStyles = makeStyles(styles);

export default function ShopTheStory(props) {
  const { products } = props;
  const classes = useStyles();

  return (
    <div className={classes.section}>
      <GridContainer justify="center">
        <GridItem xs={12} sm={10} md={10}>
          <h3 className={classes.title}>Shop The Story</h3>

          <GridContainer>
            {
              products.map (
                (product, i) =>
                  <GridItem md={4} sm={4} key={i}>
                    <Card plain product>
                      <CardHeader noShadow image>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <img src={product.img} className={classes.productImage} alt=".." />
                        </a>
                      </CardHeader>
                      <CardBody plain>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <h4 className={classes.cardTitle}>{product.brand}</h4>
                        </a>
                        <p className={classes.description}>
                          {product.description}
                        </p>
                        <div className={classes.priceContainer}>
                          <span className={classes.price}> {product.price}</span>
                        </div>
                      </CardBody>
                    </Card>
                  </GridItem>
              )
            }
          </GridContainer>
          <br />
        </GridItem>
      </GridContainer>
    </div>
  );
}

ShopTheStory.propTypes = {
  products: PropTypes.array,
};
