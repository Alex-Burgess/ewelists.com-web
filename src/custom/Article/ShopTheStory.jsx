import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
// core components
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardBody from "components/Card/CardBody.jsx";

import shopTheStoryStyle from "assets/jss/custom/components/shopTheStoryStyle.jsx";

function ShopTheStory(props) {
  const { classes, products } = props;

  return (
    <div className={classes.section}>
      <GridContainer justify="center">
        <GridItem xs={12} sm={10} md={10}>
          <h3 className={classes.title}>Shop The Story</h3>

          <GridContainer>
            {
              products.map (
                (product, i) =>
                  <GridItem md={4} sm={4} key={i}>
                    <Card plain product>
                      <CardHeader noShadow image>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <img src={product.img} className={classes.productImage} alt=".." />
                        </a>
                      </CardHeader>
                      <CardBody plain>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <h4 className={classes.cardTitle}>{product.brand}</h4>
                        </a>
                        <p className={classes.description}>
                          {product.description}
                        </p>
                        <div className={classes.priceContainer}>
                          <span className={classes.price}> {product.price}</span>
                        </div>
                      </CardBody>
                    </Card>
                  </GridItem>
              )
            }
          </GridContainer>
          <br />
        </GridItem>
      </GridContainer>
    </div>
  );
}

ShopTheStory.propTypes = {
  classes: PropTypes.object.isRequired,
  products: PropTypes.array,
};

export default withStyles(shopTheStoryStyle)(ShopTheStory);
