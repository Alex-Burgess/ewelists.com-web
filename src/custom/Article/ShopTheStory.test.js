import * as React from 'react';
import renderer from 'react-test-renderer';
import ShopTheStory from './ShopTheStory';

test('Renders SimilarArticles component.', () => {
  const storyProducts = [
    {brand: 'John Lewis & Partners', url: 'https://www.johnlewis.com/john-lewis-partners-safari-hooded-towel-and-mitt-white/p3382355', price: '£16.50', description: 'Safari Hooded Towel and Mitt, White.', img: 'https://johnlewis.scene7.com/is/image/JohnLewis/237170008?$rsp-pdp-port-640$'},
    {brand: 'Aveeno Baby', url: 'https://www.amazon.co.uk/Aveeno-Baby-Daily-Care-Cleansing/dp/B01IW7YMDS', price: '£8.04', description: 'Daily Care Cleansing Milk, 300 ml.', img: 'https://images-na.ssl-images-amazon.com/images/I/81rlToC4wtL._SX466_.jpg'}
  ];

  const tree = renderer
    .create(
      <ShopTheStory
        products={
          storyProducts
        }
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
