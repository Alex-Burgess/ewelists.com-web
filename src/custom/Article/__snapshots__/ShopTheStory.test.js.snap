// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SimilarArticles component. 1`] = `
<div
  className="makeStyles-section-1"
>
  <div
    className="MuiGrid-root makeStyles-grid-8  MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
    >
      <h3
        className="makeStyles-title-2"
      >
        Shop The Story
      </h3>
      <div
        className="MuiGrid-root makeStyles-grid-8  MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
        >
          <div
            className="makeStyles-card-113 makeStyles-cardPlain-114 makeStyles-cardProduct-121"
          >
            <div
              className="makeStyles-cardHeader-128 makeStyles-cardHeaderImage-130 makeStyles-noShadow-131"
            >
              <a
                href="https://www.johnlewis.com/john-lewis-partners-safari-hooded-towel-and-mitt-white/p3382355"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=".."
                  className="makeStyles-productImage-3"
                  src="https://johnlewis.scene7.com/is/image/JohnLewis/237170008?$rsp-pdp-port-640$"
                />
              </a>
            </div>
            <div
              className="makeStyles-cardBody-140 makeStyles-cardBodyPlain-142"
            >
              <a
                href="https://www.johnlewis.com/john-lewis-partners-safari-hooded-towel-and-mitt-white/p3382355"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4
                  className="makeStyles-cardTitle-4"
                >
                  John Lewis & Partners
                </h4>
              </a>
              <p
                className="makeStyles-description-5"
              >
                Safari Hooded Towel and Mitt, White.
              </p>
              <div
                className="makeStyles-priceContainer-6"
              >
                <span
                  className="makeStyles-price-7"
                >
                   
                  £16.50
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
        >
          <div
            className="makeStyles-card-113 makeStyles-cardPlain-114 makeStyles-cardProduct-121"
          >
            <div
              className="makeStyles-cardHeader-128 makeStyles-cardHeaderImage-130 makeStyles-noShadow-131"
            >
              <a
                href="https://www.amazon.co.uk/Aveeno-Baby-Daily-Care-Cleansing/dp/B01IW7YMDS"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=".."
                  className="makeStyles-productImage-3"
                  src="https://images-na.ssl-images-amazon.com/images/I/81rlToC4wtL._SX466_.jpg"
                />
              </a>
            </div>
            <div
              className="makeStyles-cardBody-140 makeStyles-cardBodyPlain-142"
            >
              <a
                href="https://www.amazon.co.uk/Aveeno-Baby-Daily-Care-Cleansing/dp/B01IW7YMDS"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4
                  className="makeStyles-cardTitle-4"
                >
                  Aveeno Baby
                </h4>
              </a>
              <p
                className="makeStyles-description-5"
              >
                Daily Care Cleansing Milk, 300 ml.
              </p>
              <div
                className="makeStyles-priceContainer-6"
              >
                <span
                  className="makeStyles-price-7"
                >
                   
                  £8.04
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>
</div>
`;
